com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/UniformBufferCollection.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace UnityEditor.ShaderFoundry
+{
+    internal class UniformBufferCollection
+    {
+        internal class UniformBufferObject
+        {
+            readonly public UniformDataSource DataSource;
+            readonly public string BufferName;
+
+            public ShaderBuilder Builder = new ShaderBuilder();
+
+            public UniformBufferObject(UniformDataSource dataSource, string bufferName)
+            {
+                DataSource = dataSource;
+                BufferName = bufferName;
+            }
+
+            public void AddDeclaration(ShaderBuilder outBuilder)
+            {
+                if (DataSource != UniformDataSource.Global)
+                {
+                    outBuilder.AddLine($"CBUFFER_START({BufferName})");
```

Comment:
I think we need to check that buffer name is not empty at this point since we allow custom ones.

Reply:
This is an object that represents a specific buffer. There should be no 'None' buffer that gets created.